src/MCPServer/lib/archivematicaMCP.py

```diff
@@ -138,57 +144,50 @@ def findOrCreateSipInDB(path, waitSleep=dbWaitSleep, unit_type='SIP'):
 
 @log_exceptions
 @auto_close_db
-def createUnitAndJobChain(path, config, terminate=False):
+def createUnitAndJobChain(path, watched_dir, workflow):
     path = unicodeToStr(path)
     if os.path.isdir(path):
         path = path + "/"
-    logger.debug('Creating unit and job chain for %s with %s', path, config)
+    logger.debug('Starting chain for %s', path)
+    if not os.path.exists(path):
+        return
     unit = None
+    unit_type = watched_dir["unit_type"]
     if os.path.isdir(path):
```

Comment:
I want this to be a normal code review. I'll try to clean up `archivematicaMCP.createUnitAndJobChain` as a separate commit.

Reply:
Hey so I've tried but I'm actually super worried to introduce new behaviour. What were you exactly thinking? Can you show me some code? Were you thinking on something like the following?

```python
@log_exceptions
@auto_close_db
def createUnitAndJobChain(path, watched_dir, workflow):
    path = unicodeToStr(path)
    logger.debug('Starting chain for %s', path)
    if not os.path.exists(path):
        return
    unit_type = watched_dir["unit_type"]
    if unit_type not in ("SIP", "DIP", "Transfer"):
        raise ValueError("Unexpected ``unit_type``")
    if unit_type == "Transfer":
        unit = unitTransfer(path)
        jobChain(unit, watched_dir.chain, workflow)
        return
    elif not os.path.isdir(path):
        return
    path = os.path.join(path, "")
    unit = getattr(sys.modules[__name__], "unit{}".format(unit_type))(
        path, findOrCreateSipInDB(path, unit_type=unit_type)
    )
    jobChain(unit, watched_dir.chain, workflow)
```

Reply:
That's really nice! I couldn't do it in as few lines :+1: 

I took yours and just removed the `ValueError` as those values are required in the schema. So we end up with:

``` python
    @log_exceptions
    @auto_close_db
    def createUnitAndJobChain(path, watched_dir, workflow):
        path = unicodeToStr(path)
        logger.debug('Starting chain for %s', path)
        if not os.path.exists(path):
            return
        unit_type = watched_dir["unit_type"]
        # Transfers can be files or directories.
        if unit_type == "Transfer":
            unit = unitTransfer(path)
            jobChain(unit, watched_dir.chain, workflow)
            return
        # Otherwise, we're only taking directories.
        elif not os.path.isdir(path):
            return
        path = os.path.join(path, "")
        unit = getattr(sys.modules[__name__], "unit{}".format(unit_type))(
            path, findOrCreateSipInDB(path, unit_type=unit_type)
        )
        jobChain(unit, watched_dir.chain, workflow)
```

But having chatted to you offline and reading your comment here, I think, if you're uncomfortable to add the new behavior, we should leave this for now. Perhaps the separate commit is an okay balance?